src/transformers/models/paligemma/processing_paligemma.py

```diff
@@ -213,36 +255,17 @@ def __call__(
         if max_length is not None:
             max_length += self.image_seq_length  # max_length has to account for the image tokens
 
-        if tokenize_newline_separately:
-            inputs = self.tokenizer(
-                input_strings,
-                add_special_tokens=False,
-                return_tensors=None,
-                padding="do_not_pad",
-                max_length=max_length,
-                truncation=truncation,
-            )
-            newline_token = self.tokenizer.convert_tokens_to_ids("\n")
-            concatenated_ids = [ids + [newline_token] for ids in inputs["input_ids"]]
-            concatenated_attention_masks = [mask + [1] for mask in inputs["attention_mask"]]
-
-            text_inputs = self.tokenizer.pad(
-                {"input_ids": concatenated_ids, "attention_mask": concatenated_attention_masks},
-                max_length=max_length,
-                padding=padding,
-                return_tensors=return_tensors,
-            )
-        else:
-            text_inputs = self.tokenizer(
-                input_strings,
-                add_special_tokens=False,
-                return_tensors=return_tensors,
-                padding=padding,
-                max_length=max_length,
-                truncation=truncation,
-            )
+        inputs = self.tokenizer(
+            input_strings,
+            text_pair=labels,
+            return_tensors=return_tensors,
+            padding=padding,
+            max_length=max_length,
+            truncation=truncation,
+            return_token_type_ids=return_token_type_ids,
```

Comment:
maybe add_special tokens should be false here? when using the configs from `https://huggingface.co/google/paligemma-3b-pt-224/tree/main`
I get the improper input of
```
<bos><image><image>...<image><image><bos>detect fracture
<bos><loc0260><loc0419><loc0292><loc0633> fracture<eos><pad><pad><pad><pad><pad><pad><pad>
```
Not the three `<bos>` tokens, the one at the beginning of the image token sequence and at the beginning of the suffix are incorrect I believe

Reply:
( and I only get `<eos>` because I added it myself )

Reply:
yes 👍🏻

Reply:
Yes, just switched to False there :)

Reply:
you can't switch to false, 

```
ValueError: Asking to return token_type_ids while setting add_special_tokens to False results in an undefined behavior. Please set add_special_tokens to True or set return_token_type_ids to None.
```

Reply:
probably makes more sense to just force `add_eos_token` to false on the gemma tokenizer bc we're always adding it ourselves anyways

Reply:
ah? I'm not getting this error weirdly

Reply:
okay, seems to work with Fast tokenizer but not the slow version

Reply:
hmmm ok. Fast tokenizer is tokenizing `<loc0000>` correctly for you?

Reply:
yes, seems there's an issue - adding a fix in the processor

Reply:
It looks like it might be a sentencepiece issue - use the "slow" tokenizer for now...

Reply:
ah dang! bummer!

Reply:
Fixed btw